rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py

```diff
@@ -0,0 +1,719 @@
+"""
+[1] Mastering Diverse Domains through World Models - 2023
+D. Hafner, J. Pasukonis, J. Ba, T. Lillicrap
+https://arxiv.org/pdf/2301.04104v1.pdf
+
+[2] Mastering Atari with Discrete World Models - 2021
+D. Hafner, T. Lillicrap, M. Norouzi, J. Ba
+https://arxiv.org/pdf/2010.02193.pdf
+"""
+from typing import Any, Dict, Mapping, Tuple, Union
+
+import gymnasium as gym
+import numpy as np
+
+from ray.rllib.algorithms.dreamerv3.dreamerv3_learner import (
+    DreamerV3Learner,
+    DreamerV3LearnerHyperparameters,
+)
+from ray.rllib.core.rl_module.marl_module import ModuleID
+from ray.rllib.core.learner.learner import NamedParamOptimizerPairs
+from ray.rllib.core.learner.tf.tf_learner import TfLearner
+from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_tf, try_import_tfp
+from ray.rllib.utils.tf_utils import symlog, two_hot, clip_gradients
+from ray.rllib.utils.typing import ResultDict, TensorType
+
+_, tf, _ = try_import_tf()
+tfp = try_import_tfp()
+
+
+class DreamerV3TfLearner(DreamerV3Learner, TfLearner):
+    """Implements DreamerV3 losses and gradient-based update logic in TensorFlow.
+
+    The critic EMA-copy update step can be found in the `DreamerV3Learner` base class,
+    as it is framework independent.
+
+    We define 3 local tensorflow optimizers for the sub components "world_model",
+    "actor", and "critic". Each of these optimizers might ue a different learning rate,
+    epsilon parameter, and gradient clipping thresholds and procedures.
+    """
+
+    @override(TfLearner)
+    def configure_optimizer_for_module(
+        self, module_id: ModuleID, hps: DreamerV3LearnerHyperparameters
+    ) -> NamedParamOptimizerPairs:
+
+        """Create the 3 optimizers for Dreamer learning: world_model, actor, critic."""
+        named_param_optimizer_pairs: NamedParamOptimizerPairs = {}
+
+        dreamerv3_module = self._module[module_id]
+
+        # World Model optimizer.
+        optim_world_model = tf.keras.optimizers.Adam(
+            learning_rate=hps.world_model_lr, epsilon=1e-8
```

Comment:
True, but I wanted to make at least the most obvious HP choices configurable. But in principal, you are right. Users should not really touch the config. But we'll see how well this will really work on envs other than those described in the paper. We still need to generate more prove that it's really a non-tuning-requiring algo :)